### ISSUE TITLE
Unpublished entries are being sent in REST & GraphQL API responses

### ISSUE THREAD

**Issue by RafidMuhymin** [strapi/strapi#16906]
### Required System information

- Node.js version: `16.20.0`
- NPM version: `8.19.4`
- Strapi version: `4.9.0`
- Database: `sqlite`
- Operating system: `Ubuntu 22.04.2 LTS`
- Is your project Javascript or Typescript: `JavaScript`

### Describe the bug

Strapi doesn't send draft entries in API responses by default until it gets published. But if I *unpublish* an entry after publishing it, it's still included in both the REST & GraphQL API responses. Restarting the server doesn't solve this issue.

### Steps to reproduce the behavior

1. Publish an entry
2. Then unpublish the entry
3. Notice that the unpublished entry is still included in the API responses.

### Expected behavior

It should no longer be included in the API responses after unpublishing it. 

### Screenshots

N/A

### Code snippets

N/A

### Additional context

N/A

**Comment by cdpark0530** [comment]
What I found is that this doesn't happen (at least with graphql) with queries for a collection, but does happen with queries for an entry by id.

**Comment by graphicfox** [comment]
The same happens for me. If a relation is made to a published entry and the same entry gets unpublished later on. The unpublished entry will also still be sent. 

Is there any update on this?

Node 18
Strapi 4.11.1

**Comment by Toshiro00** [comment]
Yeah same here, I created components and now I can see those data both publish and draft mode.

Also, I can't publish with checkbox it returns error:

> Please make sure all fields are valid before publishing (required field, min/max character limit, etc.)

I have to go inside of post and click "Publish".

Collection type features:

Component with single JSON field and my JSON field like this:

`[
  {
    "key": "Brand",
    "value": "Huawei"
  },
  {...},
]`

I don't know if that errors or bugs related each other.

Node 18
Strapi 4.11.2

**Comment by SakarDev** [comment]
## Happened to me also and here's the workaround:
- Using a straightforward `find` query like this works perfectly and filters out unpublished entries:
 `api/resources?fields=title`  ✅
 
- But, when fetching more details with a `findOne` and deep populating, unpublished entries are being included:
`api/resources/${id}?populate[nested_resources][populate]=description` ❌

- To fix `findOne` i had to change it to:
`api/resources/${id}?populate[nested_resources][populate]=description&populate[nested_resources][publicationState]=live`✅
